state/cluster/root_block.go

```diff
@@ -13,17 +14,24 @@ func CanonicalClusterID(epoch uint64, participants flow.IdentityList) flow.Chain
 	return flow.ChainID(fmt.Sprintf("cluster-%d-%s", epoch, participants.Fingerprint()))
 }
 
+var cachedHash flow.Identifier
+var once sync.Once
+
 // CanonicalRootBlock returns the canonical root block for the given
 // cluster in the given epoch. It contains an empty collection referencing
 func CanonicalRootBlock(epoch uint64, participants flow.IdentityList) *cluster.Block {
 
 	chainID := CanonicalClusterID(epoch, participants)
 	payload := cluster.EmptyPayload(flow.ZeroID)
+
+	once.Do(func() {
+		cachedHash = payload.Hash()
+	})
```

Comment:
I think we could even pre-calculate the payload hash for the root block:

```go
var rootBlockPayload = cluster.EmptyPayload(flow.ZeroID)
var rootBlockPayloadHash = rootBlockPayload.Hash()
```

Reply:
I like it.  thx.